atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/SweeperServiceImplTest.java

```diff
@@ -75,73 +88,76 @@ public void sweepingNonExistingTableShouldNotBeSuccessful() {
+                Optional.of(1000), Optional.of(500));
     }
 
     @Test
     public void sweepTableFromStartRowWithBatchConfigWithExactlyOneNonNullBatchConfigShouldBeSuccessful() {
-        setupTaskRunner(Mockito.mock(SweepResults.class));
-
-        when(kvs.getAllTableNames()).thenReturn(ImmutableSet.of(TABLE_REF));
-        sweeperService.sweepTableFromStartRowWithBatchConfig(TABLE_REF.getQualifiedName(),
-                encodeStartRow(new byte[] {1, 2, 3}), null, 10, null);
+        sweeperService.sweepTable(TABLE_REF.getQualifiedName(),
+                Optional.of(encodeStartRow(new byte[] {1, 2, 3})), Optional.empty(), Optional.of(10), Optional.empty(),
+                Optional.empty());
     }
 
     @Test
     public void testWriteProgressOrPriorityOrMetricsNotUpdatedAfterSweepRunsSuccessfully() {
-        setupTaskRunner(Mockito.mock(SweepResults.class));
-        when(kvs.getAllTableNames()).thenReturn(ImmutableSet.of(TABLE_REF));
-        sweeperService.sweepTableFromStartRow(TABLE_REF.getQualifiedName(), encodeStartRow(new byte[] {1, 2, 3}));
-        Mockito.verify(priorityStore, never()).update(Mockito.any(), Mockito.any(), Mockito.any());
-        Mockito.verify(progressStore, never()).saveProgress(Mockito.any(), Mockito.any());
+        sweeperService.sweepTableFrom(TABLE_REF.getQualifiedName(), encodeStartRow(new byte[] {1, 2, 3}));
+        verify(priorityStore, never()).update(any(), any(), any());
+        verify(progressStore, never()).saveProgress(any(), any());
         Mockito.verifyZeroInteractions(sweepMetrics);
     }
 
+    @Test
+    public void sweepsEntireTableByDefault() {
+        List<byte[]> startRows = ImmutableList.of(
+                PtBytes.EMPTY_BYTE_ARRAY,
+                new byte[] {0x10},
+                new byte[] {0x50});
+
+        for (int i = 0; i < startRows.size(); i++) {
+            byte[] currentRow = startRows.get(i);
+            Optional<byte[]> nextRow = (i+1) == startRows.size()
+                    ? Optional.empty()
+                    : Optional.of(startRows.get(i+1));
+
+            SweepResults results = SweepResults.createEmptySweepResult(nextRow);
+            when(sweepTaskRunner.run(any(), any(), eq(currentRow))).thenReturn(results);
+        }
+
+        sweeperService.sweepTable(TABLE_REF.getQualifiedName());
+
+        startRows.forEach(row -> verify(sweepTaskRunner).run(any(), any(), eq(row)));
+        verifyNoMoreInteractions(sweepTaskRunner);
+    }
+
+    @Test
+    public void runsOneIterationIfRequested() {
+        SweepResults resultsWithMoreToSweep = SweepResults.createEmptySweepResult(Optional.of(new byte[] {0x55}));
+        setupTaskRunner(resultsWithMoreToSweep);
+
+        sweeperService.sweepTable(TABLE_REF.getQualifiedName(), Optional.empty(), Optional.of(false),
+                Optional.empty(), Optional.empty(), Optional.empty());
+
+        verify(sweepTaskRunner, times(1)).run(any(), any(), any());
```

Comment:
nit: We should verify the results are not recorded here as well.

Reply:
done